### PR TITLE
WuJinhan1 iP

### PR DESCRIPTION
**Pull request by WuJinhan1**
# Jard
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Jard frees your mind of having to remember things you need to do. It's,

+ text-based
+ easy to learn
+ ~~FAST~~ _SUPER_ FAST to use
+ All you need to do is,

download it from [here](https://nus-cs2103-ay2425s1.github.io/website/schedule/week4/project.html).
1. double-click it.
2. add your tasks.
3. let it manage your tasks for you 🧌
4. And it is __FREE__!

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

### CODE REVIEW

File: src/main/java/Jard/Jard.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+     * entering the main loop to process user commands.
+     */
+    public void run() {
+        ui.showWelcome();
+        while (true) {
+            String input = ui.readCommand();
+            String[] inputParts = Parser.parseCommand(input);
+            String command = inputParts[0];
+
+            try {
```
Comment (jevan13) [inline src/main/java/Jard/Jard.java]: Should all these operations be done in a separate class, eg. TaskList class?

File: src/main/java/Jard/Parser.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
+package Jard;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * The Parser class is responsible for parsing user input commands, tasks, and dates.
+ * It converts user input strings into corresponding task objects or other necessary data types.
+ */
+public class Parser {
```
Comment (jevan13) [inline src/main/java/Jard/Parser.java:10]: LGTM!

File: src/main/java/Jard/Storage.java [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.FileNotFoundException;
+
+/**
+ * The Storage class handles the reading and writing of task data to and from a file.
+ * It is responsible for loading tasks from a file upon startup and saving tasks to a file when changes are made.
+ */
+public class Storage {
```
Comment (jevan13) [inline src/main/java/Jard/Storage.java:15]: LGTM!

File: src/main/java/Jard/Task.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+     * @return The status icon of the task.
+     */
+    public String getStatusIcon() {
+        return (isDone ? "X" : " ");
+    }
+
+    /**
+     * Marks the task as done.
+     */
+    public void markAsDone() {
```
Comment (jevan13) [inline src/main/java/Jard/Task.java:45]: should markAsDone() and markAsNotDone() can be combined as one method that takes in a boolean?

File: src/main/java/Jard/Ui.java [hunk 5]
```diff
@@ -0,0 +1,155 @@
+package Jard;
+
+import java.util.Scanner;
+import java.util.List;
+
+/**
+ * Handles user interactions and displays messages to the user.
+ */
+public class Ui {
```
Comment (jevan13) [inline src/main/java/Jard/Ui.java:8]: LGTM!

File: src/main/java/Jard/Task.java [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+        this.description = description;
+        this.isDone = false;
+    }
+
+    /**
+     * Returns the description of the task.
+     *
+     * @return The description of the task.
+     */
+    public String getDescription() {
```
Comment (jevan13) [inline src/main/java/Jard/Task.java:28]: Is there a need to repeat this for all child classes?

**Review by jevan13 — COMMENTED** [review]
Looks good.
Some suggestions for your consideration.

File: src/main/java/Jard/Jard.java
Comment (thony-ui) [inline src/main/java/Jard/Jard.java:1]: You can consider making the operations done in a separate class, for example ```class execute```

File: src/main/java/Jard/Parser.java
Comment (thony-ui) [inline src/main/java/Jard/Parser.java:1]: Avoid having nested if statements to make the code more readable

File: src/main/java/Jard/Task.java
Comment (thony-ui) [inline src/main/java/Jard/Task.java:1]: Good use of abstract class

File: src/test/java/Jard/DeadlineTest.java
Comment (thony-ui) [inline src/test/java/Jard/DeadlineTest.java:1]: Should test for edge cases. Whats the expected output if its the wrong output etc
